katdal/test/test_chunkstore_s3.py

```diff
@@ -488,11 +489,10 @@ def _tweak_store_args(cls, url, authenticate=True, **kwargs):
             cls.proxy_url = 'http://{}:{}'.format(proxy_host, proxy_port)
         elif url != cls.httpd.target:
             raise RuntimeError('Cannot use multiple target URLs with http proxy')
-        url = urllib.parse.urljoin(cls.proxy_url, BUCKET + '/')
-        # The token only authorises the one known bucket
-        token = encode_jwt({'alg': 'ES256', 'typ': 'JWT'}, {'prefix': [BUCKET]})
+        # The token authorises the standard bucket and anything starting with PREFIX
+        token = encode_jwt({'alg': 'ES256', 'typ': 'JWT'}, {'prefix': [BUCKET, PREFIX]})
         kwargs.setdefault('token', token)
-        return url, kwargs
+        return super().prepare_store_args(cls.proxy_url, credentials=None, **kwargs)
```

Comment:
Are there potentially issues if kwargs explicitly sets credentials too? Maybe it should be `kwargs['credentials'] = None` (or `kwargs.setdefault('credentials', None)`). I guess it's not too serious since this is test code and hence by definition if there was a problem the tests would pick it up.

Reply:
It *should* be an error because you cannot have both a token and credentials (see `katdal.chunkstore_s3._auth_factory`). The tests that involve non-standard credentials will typically be on the non-token `TestS3ChunkStore` side only (like `test_public_read`).